contracts/resolvers/profiles/DNSResolver.sol

```diff
@@ -18,24 +18,24 @@ abstract contract DNSResolver is
     // A zone hash is an EIP-1577 content hash in binary format that should point to a
     // resource containing a single zonefile.
     // node => contenthash
-    mapping(bytes32 => bytes) private zonehashes;
+    mapping(uint64 => mapping(bytes32 => bytes)) private versionable_zonehashes;
 
     // Version the mapping for each zone.  This allows users who have lost
     // track of their entries to effectively delete an entire zone by bumping
     // the version number.
     // node => version
-    mapping(bytes32 => uint256) private versions;
```

Comment:
Can you replace the version mechanism here with the new one, rather than stacking them?